products/billingbudget/api.yaml

```diff
@@ -57,165 +57,217 @@ objects:
-                    - :FORECASTED_SPEND
+              A set of subaccounts of the form billingAccounts/{account_id},
+              specifying that usage from only this set of subaccounts should
+              be included in the budget. If a subaccount is set to the name of
+              the parent account, usage from the parent account will be included.
+              If the field is omitted, the report will include usage from the parent
+              account and all subaccounts, if they exist.
+            item_type: Api::Type::String
+            at_least_one_of:
+              - budget_filter.0.projects
+              - budget_filter.0.credit_types_treatment
+              - budget_filter.0.services
+              - budget_filter.0.subaccounts
+              - budget_filter.0.labels
+          - !ruby/object:Api::Type::KeyValuePairs
+            name: 'labels'
+            description: |
+              A single label and value pair specifying that usage from only
+              this set of labeled resources should be included in the budget.
+            at_least_one_of:
+              - budget_filter.0.projects
+              - budget_filter.0.credit_types_treatment
+              - budget_filter.0.services
+              - budget_filter.0.subaccounts
+              - budget_filter.0.labels
+      - !ruby/object:Api::Type::NestedObject
+        name: amount
+        required: true
+        description: |
+          The budgeted amount for each usage period.
+        properties:
           - !ruby/object:Api::Type::NestedObject
-            name: allUpdatesRule
+            name: specifiedAmount
+            exactly_one_of:
+              - amount.0.specified_amount
+              - amount.0.last_period_amount
             description: |
-              Defines notifications that are sent on every update to the
-              billing account's spend, regardless of the thresholds defined
-              using threshold rules.
+              A specified amount to use as the budget. currencyCode is
+              optional. If specified, it must match the currency of the
+              billing account. The currencyCode is provided on output.
             properties:
               - !ruby/object:Api::Type::String
-                name: pubsubTopic
-                at_least_one_of:
-                  - budget.0.all_updates_rule.0.pubsub_topic
-                  - budget.0.all_updates_rule.0.monitoring_notification_channels
+                name: currencyCode
                 description: |
-                  The name of the Cloud Pub/Sub topic where budget related
-                  messages will be published, in the form
-                  projects/{project_id}/topics/{topic_id}. Updates are sent
-                  at regular intervals to the topic.
+                  The 3-letter currency code defined in ISO 4217.
               - !ruby/object:Api::Type::String
-                name: schemaVersion
-                default_value: "1.0"
```

Comment:
Make sure to preserve this default!

Reply:
I was getting weird diff behaviour when that field wasn't sent to computed and a default value was supplied.

Reply:
What kind of weird behaviour?

Reply:
this happens with default of 1.0

```
=== CONT  TestAccBillingBudget_billingBudgetNotifyExample
    provider_test.go:266: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_billing_budget.budget will be updated in-place
          ~ resource "google_billing_budget" "budget" {
                billing_account = "REDACTED"
                display_name    = "Example Billing Budget9unz3t7jkl"
                id              = "billingAccounts/REDACTED/budgets/c8d77e04-4231-4c75-829f-2eecfc7eb447"
                name            = "billingAccounts/REDACTED/budgets/c8d77e04-4231-4c75-829f-2eecfc7eb447"
        
              ~ all_updates_rule {
                    disable_default_iam_recipients   = true
                    monitoring_notification_channels = [
                        "projects/REDACTEDnotificationChannels/12763365363224154349",
                    ]
                  + schema_version                   = "1.0"
                }
        
                amount {
                    last_period_amount = false
        
                    specified_amount {
                        currency_code = "USD"
                        nanos         = 0
                        units         = "100000"
                    }
                }
        
                budget_filter {
                    credit_types           = []
                    credit_types_treatment = "INCLUDE_ALL_CREDITS"
                    labels                 = {}
                    projects               = [
                        "projects/550924169191",
                    ]
                    services               = []
                    subaccounts            = []
                }
        
                threshold_rules {
                    spend_basis       = "CURRENT_SPEND"
                    threshold_percent = 1
                }
                threshold_rules {
                    spend_basis       = "FORECASTED_SPEND"
                    threshold_percent = 1
                }
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
```

Looking closer at the GET call, the API doesn't return the schema field.

```
---[ REQUEST ]---------------------------------------
GET /v1/billingAccounts/REDACTED/budgets/504fdfa4-36f7-47af-a7d7-50d0877b184b?alt=json HTTP/1.1
Host: billingbudgets.googleapis.com
User-Agent: Terraform/0.13.2 (+https://www.terraform.io) Terraform-Plugin-SDK/2.1.0 terraform-provider-google-beta/acc
Content-Type: application/json
X-Goog-User-Project: REDACTED
Accept-Encoding: gzip


-----------------------------------------------------
2020/12/04 20:33:08 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Transfer-Encoding: chunked
Alt-Svc: h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Fri, 04 Dec 2020 20:33:08 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

317
{
  "name": "billingAccounts/REDACTED/budgets/504fdfa4-36f7-47af-a7d7-50d0877b184b",
  "displayName": "Example Billing Budgetay00qjnrv3",
  "budgetFilter": {
    "projects": [
"projects/550924169191"
    ],
    "creditTypesTreatment": "INCLUDE_ALL_CREDITS"
  },
  "amount": {
    "specifiedAmount": {
      "currencyCode": "USD",
      "units": "100000"
    }
  },
  "thresholdRules": [
    {
      "thresholdPercent": 1,
      "spendBasis": "CURRENT_SPEND"
    },
    {
      "thresholdPercent": 1,
      "spendBasis": "FORECASTED_SPEND"
    }
  ],
  "notificationsRule": {
    "monitoringNotificationChannels": [
"projects/REDACTED/notificationChannels/18210989568614271574"
    ],
    "disableDefaultIamRecipients": true
  },
  "etag": "1607113987289758"
}

0

```

I'm going to omit the default value.

Reply:
Can you specify https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/custom_flatten/default_if_empty.erb in that case? My concern is that the change is a breaking change in Terraform if we don't preserve it.